code/modules/events/grid_check.dm

```diff
@@ -0,0 +1,37 @@
+/datum/round_event_control/grid_check
+	name = "Grid Check"
+	typepath = /datum/round_event/grid_check
+	weight = 10
+	min_players = 3
+	earliest_start = 6000
+	max_occurrences = 3
+
+/datum/round_event/grid_check
+	announceWhen	= 1
+	startWhen = 1
+
+/datum/round_event/grid_check/announce()
+	priority_announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", 'sound/AI/poweroff.ogg')
+
+
+/datum/round_event/grid_check/start()
+	for(var/obj/machinery/power/smes/S in machines)
```

Comment:
Make a global list of SMES and track/untrack them in new/destroy, the machine list can get quite long